MinecraftClient/McClient.cs

```diff
@@ -1723,10 +1723,23 @@ public void OnSetSlot(byte inventoryID, short slotID, Item item)
             // Handle cursor item
             if (inventoryID == 255 && slotID == -1)
             {
-                if (item != null)
-                    inventories[0].Items[-1] = item;
-                else
-                    inventories[0].Items.Remove(-1);
+                if (inventories.ContainsKey(0))
+                {
+                    if (item != null)
+                        inventories[0].Items[-1] = item;
+                    else
+                        inventories[0].Items.Remove(-1);
+                }
+            }
+            else if (inventoryID == 254) // To add item to player inventory without animation

```

Comment:
To limit copy and paste, how about doing this at the beginning of function:
```C#
if (inventoryID == 254)
    inventoryID = 0;
```
Handling for inventory ID 254 is slightly different than other inventories, not sure if that's intentional.

Reply:
Done.

Reply:
> Handling for inventory ID 254 is slightly different than other inventories, not sure if that's intentional.

It should be the same except no item adding animation.

Reply:
Ok, since we do not animate this... it's the same 😄